client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -57,6 +60,13 @@ const PatternAssembler: Step = ( { navigation, flow, stepName } ) => {
 	const siteId = useSiteIdParam();
 	const siteSlugOrId = siteSlug ? siteSlug : siteId;
 	const allPatterns = useAllPatterns();
+	const sectionPatterns = useSectionPatterns();
+
+	// Fetching the categories so they are ready when ScreenCategoryList loads
+	const categoriesQuery = usePatternCategories( site?.ID );
+	const categories = ( categoriesQuery?.data || [] ) as Category[];
```

Comment:
[nit] We can let `usePatternCategories` return the categories directly. But it's optional and not a blocker.

Reply:
I tried that and ran into issues because of the typescript linter. I'll happily update it if you can show me an example.

Reply:
Maybe https://github.com/Automattic/wp-calypso/compare/feature/site-assembler-categories...feature/site-assembler-categories-example?expand=1

Reply:
Thanks! I'll try this later.

Reply:
I'm looking into this, but the link is not working. Can you share it again, please?

Reply:
https://github.com/Automattic/wp-calypso/commit/fdf667ab0b7c6598daea6ce5279ca482ff20a2c0